platform/spire/src/command.py

```diff
@@ -75,7 +75,10 @@ def invoke(args):
             fail("not enough parameters (expected %s)" % expect)
         if maxarg is not None and len(params) > maxarg:
             fail("too many parameters (expected %s)" % expect)
-        func(*params)
+        varnames = func.__code__.co_varnames
+        opts = vars(args)
```

Comment:
Are you sure this is the recommended way to do this? It seems like it would have weird edge case things, like maybe including "params" as an option, when it's really the list of parameters. Is there no easier way to do this?

Reply:
I couldn't find an easier way to do this. If it helps, as suggested, I could rename "params", "invoke", etc. so that they will be unlikely to conflict with actual function parameters.

Reply:
Apparently `vars` _is_ the recommended way to do this. Huh.